azavea/sentinel-2-stac-utils
# Create static STAC catalog from Sentinel-2 inventory

## Overview

This PR creates a STAC catalog from data like the inventory rows describing data in the Sentinel-2 buckets in AWS.

There are a few things that it doesn't do, at least yet:

- [ ] actually read a CSV (I'm using hardcoded data from a subset of one of them) (#8)
- [ ] include assets in items (#9)
- [x] include EO extension information in items
- ~include sat extension information in items~ I don't think I know enough to include [this information](https://github.com/radiantearth/stac-spec/tree/master/extensions/sat#item-fields)
- [x] include Sentinel metadata a la https://sentinel-stac.s3.amazonaws.com/sentinel-2-l1c/2/G/LS/2016-12-20/S2A_2GLS_20161220_0.json
- [x] write the root collection

### Checklist

- ~New tests have been added or existing tests have been modified~ someday

### Notes

- ~I didn't include band assets in items because http links to band assets for Sentinel-2 are 100% useless -- s2 data is in a requester pays bucket, so you have to authenticate somehow. I included a `sentinel:data_path` property in the item so you, dear user, can tack a `B02.jp2` onto the end if you want that band and make the s3 request yourself.~ we still can't do anything with the jp2s, but there's nowhere obvious to document "how to find bands from the STAC items in this catalog", so now they're included a1fac05
- There's a lot of repetitive code -- `makeFooParent` and `writeFooCatalog` both duplicate a _ton_ of stuff. I spent a little bit of effort trying to figure out how to clean that up, but I didn't dwell on it since the methods are short, the variations are small, and the first try didn't go too well. It definitely _can_ be DRYed out though.

### Testing Instructions

- `./sbt bloopInstall`
- `AWS_PROFILE=raster-foundry AWS_REGION=us-east-1 bloop run s2stac -- create --collection L1C --inventory-path bar --output-catalog-root <somewhere on s3 you have write access to>`
- sync the catalog down
- follow links from the root collection to the items and ensure that they all:
  - [x] make sense
  - [x] are symmetrical
  - [x] link to actually existing data
- download at least one of the bands from the assets on one of the items and view it with qgis
- `AWS_PROFILE=raster-foundry AWS_REGION=us-east-1 bloop run s2stac -- create --collection L1C --inventory-path bar --output-catalog-root <somewhere local>`
- check the same stuff as for s3. links:
  - [x] make sense
  - [x] are symmetrical
  - [x] link to actually existing data

- bonus round! import into Franklin -- blocked by https://github.com/azavea/franklin/issues/344

Closes #2 (if applicable)
